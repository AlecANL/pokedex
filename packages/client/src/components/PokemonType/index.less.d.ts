// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'pokemon-type-component': string;
}
declare var cssExports: CssExports;
export = cssExports;
