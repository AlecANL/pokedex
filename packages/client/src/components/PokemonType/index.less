@import (reference) '../../global.less';

.pokemon-type-component {
  width: 28px;
  height: 28px;
  background: transparent;
  display: flex;
  align-items: center;
  justify-content: center;
  .rounded(50%);
}
