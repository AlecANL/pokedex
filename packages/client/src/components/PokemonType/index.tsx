import React, { Suspense } from 'react';
import style from './index.less';

interface IPokemonTypeProps {
  /***
   * Type of pokemon
   */
  icon: PokemonType;

  /**
   * Dimension of badge
   */
  size?: number;
}

type IconTypeMap = Record<PokemonType, { color: string; image: string }>;

const typeToIconMap: IconTypeMap = {
  Bug: { color: '#74515C', image: 'bug' },
  Ground: { color: '#917464', image: 'crack' },
  Dragon: { color: '#448B95', image: 'dragon' },
  Electric: { color: '#E2E32F', image: 'electric' },
  Water: { color: '#0F7199', image: 'water' },
  Ice: { color: '#0F7199', image: 'snow' },
  Normal: { color: '#73525B', image: 'normal' },
  Fairy: { color: '#741DCA', image: 'fairy' },
  Fighting: { color: '#994025', image: 'fighting' },
  Fire: { color: '#AA1F23', image: 'fire' },
  Flying: { color: '#4A677D', image: 'fly' },
  Grass: { color: '#1A4925', image: 'leaf' },
  Poison: { color: '#9969D7', image: 'poison' },
  Psychic: { color: '#A02B6D', image: 'eye' },
  Rock: { color: '#58575E', image: 'rock' },
  Dark: { color: '#040706', image: 'moon' },
  Steel: { color: '#5F756D', image: 'hook' },
};

/**
 * This component allow to render respective "badge icon" for each pokemon type. It load dynamically and on-demand predefined icon
 * @param props
 */
const PokemonType: React.FC<IPokemonTypeProps> = props => {
  const { size, icon } = props;

  const Icon = React.lazy(() => import(`../../assets/${typeToIconMap[icon].image}`));

  return (
    <div
      style={{ backgroundColor: typeToIconMap[icon].color, width: size, height: size }}
      className={style['pokemon-type-component']}
    >
      <Suspense
        fallback={
          <div style={{ backgroundColor: '#eee' }} className={style['pokemon-type-component']} />
        }
      >
        <Icon />
      </Suspense>
    </div>
  );
};

export default PokemonType;
