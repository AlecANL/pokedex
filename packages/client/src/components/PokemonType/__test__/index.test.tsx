import React from 'react';
import renderer from 'react-test-renderer';
import PokemonType from '..';

describe('Components/PokemonType', () => {
  it('Get snapshot', () => {
    const component = renderer.create(<PokemonType icon={'Bug'} />);
    let snap = component.toJSON();
    expect(snap).toMatchSnapshot();
  });
});
