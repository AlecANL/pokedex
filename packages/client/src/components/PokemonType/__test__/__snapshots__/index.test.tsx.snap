// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/PokemonType Get snapshot 1`] = `
<div
  className="pokemon-type-component"
  style={
    Object {
      "backgroundColor": "#74515C",
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    className="pokemon-type-component"
    style={
      Object {
        "backgroundColor": "#eee",
      }
    }
  />
</div>
`;
